PowerEditor/src/ScitillaComponent/AutoCompletion.cpp

```diff
@@ -65,7 +65,20 @@ bool AutoCompletion::showApiComplete()
 
 	_pEditView->execute(SCI_AUTOCSETSEPARATOR, WPARAM(' '));
 	_pEditView->execute(SCI_AUTOCSETIGNORECASE, _ignoreCase);
-	_pEditView->showAutoComletion(curPos - startPos, _keyWords.c_str());
+
+	//determine if the file is a tex file
+	const TCHAR *fn = ((_pEditView)->getCurrentBuffer())->getFileName();
+	int fnsize = lstrlen(fn);
+	bool isTEX = fnsize >= 4 && fn[fnsize - 1] == 'x' && fn[fnsize - 2] == 'e' \
+		&& fn[fnsize - 3] == 't' && fn[fnsize - 4] == '.';
+
+	if (isTEX) { //if this is a TEX file
+		TCHAR initChars[2];
+		_pEditView->getGenericText(initChars, 2, startPos - 1, startPos);
+		if(initChars[0] == '\\')
+			_pEditView->showAutoComletion(curPos - startPos, _keyWords.c_str());
+	}else
+		_pEditView->showAutoComletion(curPos - startPos, _keyWords.c_str());

```

Comment:
You added the same code (more or less) four times to determine whether the file is a 'tex' file. Create a separate function instead e.g. `bool isTexFile( ... )`.
On the other hand it might not be necessary to create such a function as there are examples throughout the code how to determine whether special language conditions apply. Search for `getLangType` and find out how it's handled.